Documentation/design-docs/dllmap.md

```diff
@@ -0,0 +1,246 @@
+# Dllmap design document
+This document is intended to describe a plan on delivering a Dllmap feature for .NET Core.
+
+Author: Anna Aniol (@annaaniol)
+
+## Background
+
+### .NET Core P/invoke mechanism
+There is a .NET directive ([DllImport](https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.dllimportattribute(v=vs.110).aspx)) indicating that the attributed method is exposed by an unmanaged DLL as a static entry point. It enables to call a function exported from an unmanaged DLL inside a managed code. To make such a call, library name must be provided. Names of naive libraries are OS-specific, so once the name is defined, the call can be executed correctly on one OS only.
+
+Right now, CoreCLR has no good way to handle the differences between platforms when it comes to p/invoking native libraries. Here is an example usage of DllImport:
+```c++
+// Use DllImport to import the Win32 MessageBox function.
+   [DllImport("user32.dll", CharSet = CharSet.Unicode)]
+   public  static  extern  int  MessageBox(IntPtr  hWnd, String text, String caption, uint type);
+```
+This import works with Windows, but it doesn’t work with any other OS. If run e.g. on Linux, a DllNotFoundException will be thrown, which means that a DLL specified in a DLL import cannot be found.
+
+### Mono’s Dllmap
```

Comment:
I would add a little info about Mono DllMap limitation: 
1. The number of arguments must be same.
2. The argument type must be compatible(or even same).
3. No calling convention convert.